services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterServiceImpl.java

```diff
@@ -257,32 +275,21 @@ public RouteTable getSubnetRouteTable(String projectId, String subnetId) throws
         queryParams.put("owner", values);
 
         routeTableMap = this.routeTableDatabaseService.getAllRouteTables(queryParams);
-        if (routeTableMap == null) {
-            routeTableMap = new HashMap<>();
-        }
 
-        if (routeTableMap.size() == 0) {
-            // create a subnet route table
-            SubnetWebJson subnetWebJson = this.vpcRouterToSubnetService.getSubnet(projectId, subnetId);
-            String vpcId = subnetWebJson.getSubnet().getVpcId();
-            Router router = getOrCreateVpcRouter(projectId, vpcId);
+        if (routeTableMap == null || routeTableMap.size() == 0) {
+            throw new CanNotFindRouteTableByOwner();
+        }
```

Comment:
@DavidLiu506 Don't throw an exception if routetable is not exist here, just return null.

Reply:
Talked with James. Throw routertable not found.

Reply:
agree